Refactoring and send continues heartbeat

arminc opened the pull request:
I wanted to add the possibility to send continues heartbeats by a specified interval but before doing that I did some refactoring. I apologies for such a big PR but I thought you may find it interesting.
##### What is in the PR:
- Added 'github.com/codegangsta/cli' for arguments parsing and help messages
- Added 'github.com/Sirupsen/logrus' for logging
- Using 'godep' for dependencies 
- Reduced some 'DRY'
- Split code in to multiple files
- Added tests
- Added 'send continues heartbeat'
##### Things that can be improved:
- More tests
- Retry 'send heartbeat' for X time

halitokumus commented:
Thank you very much @arminc , for this collaboration. I'll take a look at this in the next week, hopefully.

arminc commented:
In this PR the app stops when an error or a timeout happens, this is not that nice when running this in production because a network hiccup should not stop the app. I have fixed this by removing the exit and trowing the errors up so that the decision if needed can be made to stop or ... I also fixed some golint warning. The changes stack on top op this PR and you can find them here: https://github.com/arminc/opsgenie-heartbeat/tree/retry 

I haven't merged them in here but I can if you want to.

cglrkn commented:
Hi,

We updated our plugin to use APIv2 and added logging capability to the plugin as you suggested. However, we couldn't accept your pull request since it is designed to use APIv1. Thank you for your collaboration.

Best Regards
